cmd/ndm_daemonset/controller/blockdevicestore.go

```diff
@@ -32,6 +32,9 @@ import (
 // blockDevice is DeviceResource-CR
 func (c *Controller) CreateBlockDevice(blockDevice apis.BlockDevice) error {
 
+	// set namespace on the api resource
```

Comment:
It is good that this is being set now. Prior to this change also, BDs were namespace scoped, was K8s automatically setting them?

Reply:
earlier we were setting the namespace while converting to the BD API resource. Now we are setting it just before creating. So that the namespace of client is provided. 

Also, we were previously not using the `GetBlockDevice()` method anywhere in our code. In that we were not using Namespace, it was given as empty. So it will cause the Get() call to error out. This was found out while working with the other PR.